sdk/iot/README.md

```diff
@@ -1,139 +1,163 @@
+
+### Prerequisites
+
+For compiling the Azure IoT Embedded C SDK for the most common platforms (Windows and Linux), no further prerequisites are necessary.
+Please follow the instructions in the Getting Started section above.
+For compiling for specific target devices, please refer to their specific toolchain documentation. 
+
+## Key Features
+
+
+Feature | Azure IoT Embedded C SDK | Description
+---------|----------|---------------------
+ [Send device-to-cloud message](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-d2c) | [Suported] | Send device-to-cloud messages to IoT Hub with the option to add custom message properties. 
+ [Receive cloud-to-device messages](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d) | [Supported] | Receive cloud-to-device messages and associated properties from IoT Hub.   
+ [Device Twins](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-device-twins) | [Supported] | IoT Hub persists a device twin for each device that you connect to IoT Hub.  The device can perform operations like get twin document, subscribe to desired property updates.
+ [Direct Methods](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-direct-methods) | [Supported] | IoT Hub gives you the ability to invoke direct methods on devices from the cloud.  
+ [DPS - Device Provisioning Service](https://docs.microsoft.com/azure/iot-dps/) | [Supported] | This SDK supports connecting your device to the Device Provisioning Service via, for example, [individual enrollment](https://docs.microsoft.com/azure/iot-dps/concepts-service#enrollment) using a [X.509 leaf certificate](https://docs.microsoft.com/azure/iot-dps/concepts-security#leaf-certificate).  
+ Protocol | MQTT | The Azure IoT Embedded C SDK suports only MQTT.  
+ Retry Policies | [Partially supported] | The Azure IoT Embedded C SDK provides guidelines for retries, but actual retries should be handled by the application.
+ [Plug and Play](https://docs.microsoft.com/en-us/azure/iot-pnp/overview-iot-plug-and-play) | [Not supported yet] | IoT Plug and Play Preview enables solution developers to integrate devices with their solutions without writing any embedded code. 
+
+
+
+
+## Examples
+
+### IoT Hub Client
+
+To use IoT Hub connectivity, the first action by a developer should be to initialize the
+client with the `az_iot_hub_client_init()` API. Once that is initialized, you may use the
+`az_iot_hub_client_get_user_name()` and `az_iot_hub_client_get_client_id()` to get the
+user name and client id to establish a connection with IoT Hub.
+
+An example use case is below.
+
+```C
+//FOR SIMPLICITY THIS DOES NOT HAVE ERROR CHECKING. IN PRODUCTION ENSURE PROPER ERROR CHECKING.
```

Comment:
Nit: These are bound to run out of date. Instead of maintaining this (which doesn't have error checking) as well as samples we should keep code in one single place.
There are some solutions to reference and embed source code from C files within the repo into the MD file although more research is required. Let's think about how to remove/replace with references in the future (for now I don't have an immediate solution.)

Reply:
Yea if anything we can have code snippets in the doxygen to go directly with the methods. Doing it there makes it easier to keep track if anything changes. Plus, when people look up the docs for those functions, examples will be right there for them. Less hunting around.